tests/parsenum/main.c

```diff
@@ -16,16 +16,36 @@ main(int argc, char * argv[])
 
 	WARNP_INIT;
 
-#define TEST(x, y, z, w) do {						\
-	fprintf(stderr, "PARSENUM(\"%s\", %s, %s)\n", y, #z, #w);	\
-	if (PARSENUM(x, y, z, w))					\
-		warnp("PARSENUM");					\
-	fprintf(stderr, "%f %zu %d\n", d, s, i);			\
+#define TEST4_PROLOGUE(str, var, min, max, target)		\
+	var parsenum_x;						\
+	int parsenum_ret;					\
+	fprintf(stderr, "Parsing \"%s\" as %s between %s and %s (incl.) yields " target "... ", str, #var, #min, #max);			\
+	parsenum_ret = PARSENUM(&parsenum_x, str, min, max);
+
+#define TEST4_SUCCESS(str, var, min, max, target) do {		\
+	TEST4_PROLOGUE(str, var, min, max, #target);		\
+	if (parsenum_ret == 0) {				\
+		if ((parsenum_x >= target) && (parsenum_x <= target)) {	\
```

Comment:
I'd go with `fabs(x - y) < 1 && fabs(x - y) <= fabs(x + y) * 0.000001`.  The compiler warning is quite right about testing for equality being dangerous... especially if you're using gcc, which has broken floating-point arithmetic.  Checking `>= && <=` might silence the warning, but it doesn't actually solve the problem.